### PR TITLE
core: RIOT's own assert macro

### PR DESCRIPTION
**Pull request by OlegHahm**
Replaces #3722 after I screwed things up in git.

### CODE REVIEW

**Comment by OlegHahm** [comment]
@kaspar030, can you ACK?

@Kijewski, took your last comment into account.

**Comment by OlegHahm** [comment]
there was no space for a verb ;)

File: cpu/msp430-common/include/cpu.h [hunk 1]
```diff
@@ -167,6 +167,16 @@ inline void __restore_context(unsigned int irqen)
...
 void msp430_cpu_init(void);
 
+/**
+ * @brief   Print the last instruction's address
+ *
+ * @todo:   Not supported
+ */
+static inline void cpu_print_last_instruction(void)
+{
+    printf("n/a");
```
Comment (cgundogan) [inline cpu/msp430-common/include/cpu.h]: puts?

File: core/include/panic.h [hunk 2]
```diff
@@ -29,6 +29,29 @@
...
+ */
+typedef enum {
+    PANIC_GENERAL_ERROR,
+    PANIC_SOFT_REBOOT,
+    PANIC_HARD_REBOOT,
+    PANIC_ASSERT_FAIL,
+#ifdef MODULE_CORTEXM_COMMON
+    PANIC_NMI_HANDLER,       /**< non maskable interrupt */
+    PANIC_HARD_FAULT,        /**< hard fault */
+#if defined(CPU_ARCH_CORTEX_M3) || defined(CPU_ARCH_CORTEX_M4) || \
```
Comment (cgundogan) [inline core/include/panic.h]: parens

File: cpu/msp430-common/include/cpu.h (hunk 1 above)
Comment (OlegHahm) [inline cpu/msp430-common/include/cpu.h]: Hm, I used printf() in order to allow for embedding this output.

File: core/include/panic.h (hunk 2 above)
Comment (OlegHahm) [inline core/include/panic.h]: Outside from gnrc it's always note like this.

File: cpu/lpc2387/include/cpu.h [hunk 3]
```diff
@@ -16,6 +16,7 @@
  * @{
  */
 
+#include <stdio.h>
 #include <stdbool.h>
 #include "lpc2387.h"
```
Comment (cgundogan) [inline cpu/lpc2387/include/cpu.h]: new line

**Comment by OlegHahm** [comment]
Updated to make Travis and @cgundogan happy.

**Comment by cgundogan** [comment]
tested on native and on the iotlab-m3. ACK

**Comment by OlegHahm** [comment]
squashed - @kaspar030, does your ACK hold?

**Comment by OlegHahm** [comment]
Fixed Travis warnings and squashed immediately (missing documentation and missing include).

File: cpu/x86/panic.c [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @author      Oliver Hahm <oliver.hahm@inria.fr>
+ */
+
+#include "x86_reboot.h"
+#include "lpm.h"
+
+void panic_arch(void)
+{
+#if DEVELHELP
```
Comment (kaspar030) [inline cpu/x86/panic.c]: ifdef

**Comment by kaspar030** [comment]
ACK after last comment is addressed.

**Comment by OlegHahm** [comment]
Fixed all occurrences of this broken check.

File: cpu/native/include/cpu.h [hunk 5]
```diff
@@ -28,6 +30,16 @@ extern "C" {
 void dINT(void);
 void eINT(void);
 
+/**
+ * @brief   Prints the last instruction's address
+ */
+static inline void cpu_print_last_instruction(void)
+{
+    void *p;
+    __asm__("1: mov 1b, %0" : "=r" (p));
```
Comment (LudwigKnuepfer) [inline cpu/native/include/cpu.h]: This broke native on arm support..

Comment (LudwigKnuepfer) [inline cpu/native/include/cpu.h]: I guess not terribly many people use native on arm ;)

**Pull request merged**